webapp/src/test/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchServiceTest.java

```diff
@@ -56,32 +75,37 @@ public void itShouldUseRepositoryToFindAllStudentGroupBatches() throws IOExcepti
 
     @Test
     public void itShouldUpload() throws IOException {
-        when(file.getBytes()).thenReturn(("school_natural_id,subject_code,group_name,group_user_login,school_year,student_ssid\nschool_natural_id_val,subject_code_val,group_name_val,group_user_login_val,school_year_val,student_ssid_val\n").getBytes());
+        final String payload = payload(ValidHeaders, ValidValues);
+        when(file.getInputStream()).thenReturn(new ByteArrayInputStream(payload.getBytes()));
         final StudentGroupBatch batch = StudentGroupBatch.builder()
                 .creator(user.getUsername())
                 .status(ImportStatus.ACCEPTED)
-                .digest(DigestUtils.md5Hex(file.getBytes())
+                .digest(DigestUtils.md5Hex(payload.getBytes())
                         .toUpperCase())
```

Comment:
This proves that the DigestInputStream produces the same digest as calling it against a byte array.